odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,546 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Operators of linearized deformations.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator.operator import Operator
+from odl import ProductSpace, DiscreteLp, Gradient, PointwiseInner, Divergence
+import numpy as np
+
+
+__all__ = ('LinDeformFixedTempl', 'LinDeformFixedTemplDeriv',
+           'LinDeformFixedTemplDerivAdj', 'LinDeformFixedDisp',
+           'LinDeformFixedDispAdj', 'linear_deform', 'deform_grad')
+
+
+def linear_deform(template, displacement):
+    """Linearized deformation of a template with a displacement field.
+
+    The template ``I`` is deformed by a displacement field ``v`` as:
+
+        (I, v) --> I(Id + v)
+
+    where ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain.
```

Comment:
This whole mathematical description should IMO go into a `Notes` section with proper LaTeX markup. I'd write it like this:
```
    Notes
    -----
    A template :math:`I:\Omega \\to \mathbb{R}` defined on
    :math:`\Omega\subset\mathbb{R}^d` is deformed by a displacement
    field :math:`v:\Omega \\to \mathbb{R}^d` according to

    .. math::

        (I, v) \mapsto I \circ (\mathrm{id} + v), \quad
        [I \circ (\mathrm{id} + v)](x) = I(x + v(x)),

    where :math:`\mathrm{id}` is the identity mapping.
```